### PR TITLE
Make output url more relative

### LINKED ISSUE #566: Heroku support?
Hi,

Has anyone tried to run this on Heroku (or similar)?

Did have time to test it myself, just curious :)

Amazing project by the way!

### PR DESCRIPTION
**Pull request by drewpc**
This fixes #566 by making the output url use '//' instead of hardcoding it to 'http://' or 'https://'.  Also adds tests for http and https requests.

### CODE REVIEW

File: packages/react-server-cli/src/run.js [hunk 1]
```diff
@@ -30,7 +29,11 @@ export default function run(options = {}) {
 		// Pass. Commands need to check for routes themselves.
 	}
 
-	options.outputUrl = jsUrl || `${httpsOptions ? "https" : "http"}://${host}:${jsPort}/`;
+	// The non-jsUrl created output URL should be as relative as possible.  '//' ensures that we use the same
+	// scheme as the browser, which is important when using a fronting server that terminates HTTPS connections and
+	// proxies requests to react-server over HTTP.  Another note, is that `host` should be `0.0.0.0` to make it relative
+	// to the hostname being used.
+	options.outputUrl = jsUrl || `//${host}:${jsPort}/`;
```
Comment (roblg) [inline packages/react-server-cli/src/run.js]: > Another note, is that `host` should be `0.0.0.0` to make it relative
>	// to the hostname being used.

Whoa! Is that a thing that browsers support in general, or something special about what we're doing here?

Comment (drewpc) [inline packages/react-server-cli/src/run.js]: Nope, it's included in the `webpack-dev-server` and `webpack-hot-middleware` client packages though:

https://github.com/webpack/webpack-dev-server/blob/master/client/index.js#L113-L121

Comment (roblg) [inline packages/react-server-cli/src/run.js]: _Interesting_.

**Comment by doug-wade** [comment]
Having linked this up locally, I'm getting the following error from both the hello-world and bike-share examples:

```
dwade react-server-examples/bike-share ‹patch-make-output-url-more-relative› » npm start                

> react-server-bike-share@0.0.1 start /home/dwade/foss/react-server/packages/react-server-examples/bike-share
> npm run clean && react-server start --hot


> react-server-bike-share@0.0.1 clean /home/dwade/foss/react-server/packages/react-server-examples/bike-share
> rm -rf build __clientTemp

{ AssertionError: missing path
    at Module.require (module.js:495:3)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/dwade/foss/react-server/packages/react-server-cli/target/react-server.js:10:18)
    at Module._compile (module.js:570:32)
    at loader (/home/dwade/foss/react-server/packages/react-server-cli/node_modules/babel-register/lib/node.js:144:5)
    at Object.require.extensions.(anonymous function) [as .js] (/home/dwade/foss/react-server/packages/react-server-cli/node_modules/babel-register/lib/node.js:154:7)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
  name: 'AssertionError',
  actual: null,
  expected: true,
  operator: '==',
  message: 'missing path',
  generatedMessage: false }
```

**Comment by doug-wade** [comment]
I linked this up and started it locally, and got the following error from the hello-world example:


```
2017-01-24T22:58:24.585Z - notice: [react-server-cli.src.commands.start] Ready for requests on 0.0.0.0:3000.
TypeError: Cannot set property 'expose' of undefined
    at Object.extendApp [as extend] (/home/dwade/foss/react-server/packages/react-server-examples/hello-world/node_modules/express-state/index.js:18:25)
    at Object.module.exports [as middleware] (/home/dwade/foss/react-server/packages/react-server-examples/hello-world/node_modules/react-server/target/server/renderMiddleware.js:48:15)
    at /home/dwade/foss/react-server/packages/react-server-cli/target/commands/start.js:152:28
    at Layer.handle [as handle_request] (/home/dwade/foss/react-server/packages/react-server-cli/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/home/dwade/foss/react-server/packages/react-server-cli/node_modules/express/lib/router/index.js:312:13)
    at /home/dwade/foss/react-server/packages/react-server-cli/node_modules/express/lib/router/index.js:280:7
    at Function.process_params (/home/dwade/foss/react-server/packages/react-server-cli/node_modules/express/lib/router/index.js:330:12)
    at next (/home/dwade/foss/react-server/packages/react-server-cli/node_modules/express/lib/router/index.js:271:10)
    at cookieParser (/home/dwade/foss/react-server/node_modules/cookie-parser/index.js:56:14)
    at Layer.handle [as handle_request] (/home/dwade/foss/react-server/packages/react-server-cli/node_modules/express/lib/router/layer.js:95:5)
TypeError: Cannot set property 'expose' of undefined
    at Object.extendApp [as extend] (/home/dwade/foss/react-server/packages/react-server-examples/hello-world/node_modules/express-state/index.js:18:25)
    at Object.module.exports [as middleware] (/home/dwade/foss/react-server/packages/react-server-examples/hello-world/node_modules/react-server/target/server/renderMiddleware.js:48:15)
    at /home/dwade/foss/react-server/packages/react-server-cli/target/commands/start.js:152:28
    at Layer.handle [as handle_request] (/home/dwade/foss/react-server/packages/react-server-cli/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/home/dwade/foss/react-server/packages/react-server-cli/node_modules/express/lib/router/index.js:312:13)
    at /home/dwade/foss/react-server/packages/react-server-cli/node_modules/express/lib/router/index.js:280:7
    at Function.process_params (/home/dwade/foss/react-server/packages/react-server-cli/node_modules/express/lib/router/index.js:330:12)
    at next (/home/dwade/foss/react-server/packages/react-server-cli/node_modules/express/lib/router/index.js:271:10)
    at cookieParser (/home/dwade/foss/react-server/node_modules/cookie-parser/index.js:56:14)
    at Layer.handle [as handle_request] (/home/dwade/foss/react-server/packages/react-server-cli/node_modules/express/lib/router/layer.js:95:5)
```

what gives?

**Comment by drewpc** [comment]
Lemme check it out on my end.

**Comment by doug-wade** [comment]
@drewpc repros on master.  I'll file it separately, but let's not hold up this pr.

**Comment by drewpc** [comment]
This also fixes #649.

**Pull request merged**